v3/operator.go

```diff
@@ -0,0 +1,182 @@
+package v3
+
+import (
+	"fmt"
+)
+
+type operator int16
+
+const (
+	unknownOp operator = iota
+
+	scanOp
+
+	unionOp
+	intersectOp
+	exceptOp
+
+	innerJoinOp
+	leftJoinOp
+	rightJoinOp
+	crossJoinOp
+	semiJoinOp
+	antiJoinOp
+
+	projectOp
+	selectOp
+
+	groupByOp
+	orderByOp
+	distinctOp
+
+	variableOp
+	constOp
+
+	existsOp
+
+	andOp
+	orOp
+	notOp
+
+	eqOp
+	ltOp
+	gtOp
+	leOp
+	geOp
+	neOp
+	inOp
+	notInOp
+	likeOp
+	notLikeOp
+	iLikeOp
+	notILikeOp
+	similarToOp
+	notSimilarToOp
+	regMatchOp
+	notRegMatchOp
+	regIMatchOp
+	notRegIMatchOp
+	isDistinctFromOp
+	isNotDistinctFromOp
+	isOp
+	isNotOp
+	anyOp
+	someOp
+	allOp
+
+	bitandOp
+	bitorOp
+	bitxorOp
+	plusOp
+	minusOp
+	multOp
+	divOp
+	floorDivOp
+	modOp
+	powOp
+	concatOp
+	lShiftOp
+	rShiftOp
+
+	unaryPlusOp
+	unaryMinusOp
+	unaryComplementOp
+)
+
+type operatorInfo struct {
+	name             string
+	columns          func(e *expr) []bitmapIndex
+	updateProperties func(e *expr)
+}
+
+func scalarInfo(name string) operatorInfo {
+	return operatorInfo{
+		name: name,
+		columns: func(expr *expr) []bitmapIndex {
+			return nil
+		},
+		updateProperties: func(expr *expr) {
+			// For a scalar operation the required input variables is the union of the
+			// required input variables of its inputs. There are no output variables.
+			expr.inputVars = 0
+			expr.outputVars = 0
+			for _, input := range expr.inputs() {
+				expr.inputVars |= input.inputVars
+			}
+		},
+	}
+}
+
+var operatorTab = [...]operatorInfo{
```

Comment:
@albler Notice how the `operatorInfo` for a particular operator is accessed via `expr.info()`. There is a minor bit of centralization here, but it isn't too bad. I think the centralization could be removed with a bit more effort.